docs/excel/excel-add-ins-ranges-advanced.md

```diff
@@ -0,0 +1,166 @@
+
+*After the preceding function has been run.*
+
+![Data in Excel after range’s copy method has been run](../images/excel-range-copyfrom-skipblanks-after.png)
+
+`transpose` determines whether or not the data is transposed, meaning its rows and columns are switched, into the source location.
+A transposed range is flipped along the main diagonal, so rows **1**, **2**, and **3** will become columns **A**, **B**, and **C**.
+
+## Remove duplicates
+
+> [!NOTE]
+> The Range object's `removeDuplicates` function is currently available only in public preview (beta). To use this feature, you must use the beta library of the Office.js CDN: https://appsforoffice.microsoft.com/lib/beta/hosted/office.js.
+> If you are using TypeScript or your code editor uses TypeScript type definition files for IntelliSense, use https://appsforoffice.microsoft.com/lib/beta/hosted/office.d.ts.
+
+The Range object's `removeDuplicates` function removes rows with duplicate entries in the specified columns. The function goes through each row in the range from the lowest-valued index to the highest-valued index in the range (from top to bottom). A row is deleted if it has a value in the specified column or columns appeared earlier in the range. Rows in the range below the deleted row are shifted up. `removeDuplicates` does not affect the position of cells outside of the range.
+
+`removeDuplicates` takes in a `number[]` representing the column indices which are checked for duplicates. This array is zero-based and relative to the range, not the worksheet. The function also takes in a boolean parameter that specifies whether the first row is a header. When **true**, the top row is ignored when considering duplicates. The `removeDuplicates` function returns a `RemoveDuplicatesResult` object that specifies the number of rows removed and the number of unique rows remaining.
+
+When using a range's `removeDuplicates` function, keep the following in mind:
+
+- `removeDuplicates` considers cell values, not function results. If two different functions evaluate to the same result, the cell values are not considered duplicates.
+- Empty cells are not ignored by `removeDuplicates`. The value of an empty cell is treated like any other value. This means empty rows contained within in the range will be included in the `RemoveDuplicatesResult`.
+
+The following sample shows the removal of entries with duplicate values in the first column.
+
+```js
+Excel.run(async (context) => {
+    var sheet = context.workbook.worksheets.getItem("Sample");
+    var range = sheet.getRange("B2:D11");
+
+    var deleteResult = range.removeDuplicates([0],true);
+    deleteResult.load();
+
+    return context.sync().then(function () {
+        console.log(deleteResult.removed + " entries with duplicate names removed.");
+        console.log(deleteResult.uniqueRemaining + " entries with unique names remain in the range.");
+});
+```
```

Comment:
I think this snippet is missing the closing `)}` corresponding to the line `return context.sync().then(function () {`.

(The `});` at the end of the snippet corresponds to the line `Excel.run(async (context) => {`.)